amqprs/src/net/channel_id_repo.rs

```diff
@@ -8,7 +8,10 @@ pub(crate) struct ChannelIdRepository {
 }
 impl ChannelIdRepository {
     pub fn new(channel_max: ShortUint) -> Self {
-        let len = 1 + (channel_max as usize - 1) / 8;
+        let len = match channel_max {
+            0 => u16::MAX as usize,
```

Comment:
```suggestion
            0 => 1 + (u16::MAX as usize - 1) / 8,
```

every byte can store 8 channel state